SoftDevSix/Argos
# refactor: migrate project to multi-module

#### Resolved task/US: ([CS-49: Configure the SonarQube to the subproject ASTProcessor](https://app.clickup.com/t/868avu2a0))
* * * *

## What's in this pull request?



1. Migrated src and ASTProcessor into a multi-module project respectively. They share logic using[ coventional plugins](https://docs.gradle.org/current/userguide/sharing_build_logic_between_subprojects.html#sec:sharing_logic_via_convention_plugins) as Gradle recommends.
2. Configured project to have an [aggregated coverage report](https://docs.gradle.org/6.4-rc-1/samples/sample_jvm_multi_project_with_code_coverage.html). This way SonarQube will perform an analisis of all the projects.


## Type of change


## How Has This Been Tested?



1. Build the project to ensure no tests are brokern `$ ./gradlew build`
2. Execute local sonar analisis to ensure the complete coverage `$ ./sonar_analysis-v1.0.sh*`

## Checklist:


- - -
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/SoftDevSix/Argos/40)
